app/grandchallenge/core/error_handlers.py

```diff
@@ -0,0 +1,98 @@
+from grandchallenge.notifications.models import Notification, NotificationType
+
+
+class ErrorHandler:
+    def handle_error(self, *, error_message, user=None, interface=None):
+        raise NotImplementedError("Subclasses must implement this method")

```

Comment:
You could use ABC here, this way everything is checked at import time rather than runtime. 

```python
class ErrorHandler(ABC):
    @abstractmethod
    def handle_error(self, *, error_message, user=None, interface=None):
        pass
```